src/sage/stats/distributions/discrete_gaussian_lattice.py

```diff
@@ -279,34 +449,103 @@ def __init__(self, B, sigma=1, c=None, precision=None):
 
             sage: v = vector(ZZ, n, (-3, -3))
             sage: v.set_immutable()
-            sage: while v not in counter:
-            ....:     add_samples(1000)
-            sage: while abs(m*f(v)*1.0/c/counter[v] - 1.0) >= 0.1:                      # needs sage.symbolic
+            sage: while v not in counter: add_samples(1000)
+            sage: while abs(m*f(v)*1.0/nf/counter[v] - 1.0) >= 0.1:                     # needs sage.symbolic
             ....:     add_samples(1000)
 
+            sage: counter = defaultdict(Integer)
             sage: v = vector(ZZ, n, (0, 0))
             sage: v.set_immutable()
             sage: while v not in counter:
             ....:     add_samples(1000)
-            sage: while abs(m*f(v)*1.0/c/counter[v] - 1.0) >= 0.1:                      # needs sage.symbolic
+            sage: while abs(m*f(v)*1.0/nf/counter[v] - 1.0) >= 0.1:                     # needs sage.symbolic
             ....:     add_samples(1000)
 
-            sage: from sage.stats.distributions.discrete_gaussian_lattice import DiscreteGaussianDistributionLatticeSampler
+        Spherical covariance are automatically handled.
```

Comment:
```suggestion
        Spherical covariance are automatically handled::
```
Also all of the ones below should have this ending as it signifies the next block is code (which is meant to be tested).